corehq/apps/dashboard/models.py

```diff
@@ -35,6 +36,15 @@ def is_visible(self):
         """Whether or not the tile is visible on the dashboard (permissions).
         :return: Boolean
         """
+        if not self.request.can_access_all_locations:
+            url = self.tile_config.get_url(self.request)
+            try:
+                match = resolve(url)
+            except Resolver404:
+                pass
+            else:
+                if 'domain' in match.kwargs and not url_is_location_safe(url):
+                    return False
```

Comment:
Note that this only affects "domain" views, so non-domain things like the help site and the exchange will appear if they'd normally appear.  "Location safety" only applies to domain things, so I figured it made sense to preserve the existing logic otherwise.